base/inc/fields/js/presets-field.js

```diff
@@ -0,0 +1,66 @@
+/* global jQuery, sowbForms */
+
+(function ( $ ) {
+	$( document ).on( 'sowsetupformfield', '.siteorigin-widget-field-type-presets', function ( e ) {
+		
+		var $presetSelect = $( this ).find( 'select[class="siteorigin-widget-input"]' );
+		if ( $presetSelect.data( 'initialized' ) ) {
+			return;
+		}
+		
+		var $undoLink = $presetSelect.find( '+ .sowb-presets-field-undo' );
+		$undoLink.hide();
+		
+		var presets = $presetSelect.data( 'presets' );
+		$presetSelect.change( function () {
+			
+			var selectedPreset = $presetSelect.val();
+			if ( selectedPreset && presets.hasOwnProperty( selectedPreset ) ) {
+				
+				var presetValues = presets[ selectedPreset ].values;
+				
+				var $formContainer = $presetSelect.closest( '.siteorigin-widget-form-main' );
+				var previousValues = $presetSelect.data( 'previousValues' );
+				if ( ! previousValues ) {
+					var presetClone = JSON.parse( JSON.stringify( presetValues ) );
+					var widgetData = sowbForms.getWidgetFormValues( $formContainer );
+					var recurseDepth = 0;
+					var copyValues = function( from, to ) {
```

Comment:
This could very well be doing something extra, but doesn't `jQuery.extend` offer this functionality already? Can we use that?

https://api.jquery.com/jquery.extend/

Reply:
Yeah, I did look at using `jQuery.extend` and it's very similar but I didn't see an option to do exactly what we need. We _only_ want the values for properties that are present on the `presets` object to be copied. `jQuery.extend` will copy all properties from source objects, even if they don't exist on the target, which could result in overwriting a user's content changes.